lib/cib/cib_ops.c

```diff
@@ -440,14 +453,7 @@ cib_process_delete(const char *op, int options, const char *section, xmlNode * r
 
     obj_root = pcmk_find_cib_element(*result_cib, section);
     if (pcmk__xe_is(input, section)) {
-        xmlNode *child = NULL;
-        for (child = pcmk__xml_first_child(input); child;
-             child = pcmk__xml_next(child)) {
-            if (replace_xml_child(NULL, obj_root, child, TRUE) == FALSE) {
-                crm_trace("No matching object to delete: %s=%s",
-                          child->name, pcmk__xe_id(child));
-            }
-        }
+        pcmk__xe_foreach_child(input, NULL, replace_child, obj_root);
 
     } else if (replace_xml_child(NULL, obj_root, input, TRUE) == FALSE) {
```

Comment:
Are you saying to just get rid of the pcmk__xe_foreach_child call and call replace_child instead?  If so, this is the only place that function is called so it could just be inlined here.

Reply:
No, the new `replace_child()` function could replace the else block

Reply:
Oh, yes, so it could.  Context can be hard to tell with github sometimes.